README.md

```diff
@@ -155,19 +158,28 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
 ### MetricsLogger
 
-The `MetricsLogger` is the interface you will use to publish embedded metrics.
+The `MetricsLogger` is the interface you will use to publish embedded metrics. 
 
-- MetricsLogger **PutMetric**(string key, double value, Unit unit)
 - MetricsLogger **PutMetric**(string key, double value)
+- MetricsLogger **PutMetric**(string key, double value, Unit unit)
+- MetricsLogger **PutMetric**(string key, double value, StorageResolution storageResolution)
+- MetricsLogger **PutMetric**(string key, double value, Unit unit, StorageResolution storageResolution)
 
-Adds a new metric to the current logger context. Multiple metrics using the same key will be appended to an array of values. The Embedded Metric Format supports a maxumum of 100 metrics per key.
+Adds a new metric to the current logger context. Multiple metrics using the same key will be appended to an array of values. Multiple metrics cannot have the same key but different storage resolutions. Same metric cannot have different storage resolutions otherwise a `InvalidMetricException` will be thrown. The Embedded Metric Format supports a maxumum of 100 metrics per key.
 
 Metrics must meet CloudWatch Metrics requirements, otherwise a `InvalidMetricException` will be thrown. See [MetricDatum](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html) for valid values.
 
+##### Storage Resolution.
+An OPTIONAL value representing the storage resolution for the corresponding metric. Setting this to High specifies this metric as a high-resolution metric, so that CloudWatch stores the metric with sub-minute resolution down to one second. Setting this to 60 specifies this metric as a Standard-resolution metric, which CloudWatch stores at 1-minute resolution. If a value is not provided, then a default value of Standard is assumed. See [Cloud Watch High-Resolution metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/publishingMetrics.html#high-resolution-metrics)
```

Comment:
```suggestion
An OPTIONAL value representing the storage resolution for the corresponding metric. Setting this to `High` specifies this metric as a high-resolution metric, so that CloudWatch stores the metric with sub-minute resolution down to one second. Setting this to `Standard` specifies this metric as a standard-resolution metric, which CloudWatch stores at 1-minute resolution. If a value is not provided, then a default value of `Standard` is assumed. See [Cloud Watch High-Resolution metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/publishingMetrics.html#high-resolution-metrics)
```